File: Data Structures & Algorithms/duplicate-integer/submission-4.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        int left = 0,  right = left + 1;
        int n = nums.length;
        while (right <= nums.length - 1) {
            if (nums[left] == nums[right]) {
                return true;
            } else {
                right++;
            }
            if (right == n) {
                left++;
                right = left + 1;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = 1
        n = len(nums)
        prefix = [1] * n

        for i in range(1, n):
            prefix[i] = res * nums[i-1]
            res = res * nums[i-1]
        
        res = 1
        postfix = [1] * n
        
        for i in range(n-2, -1, -1):
            postfix[i] = res * nums[i+1]
            res = res * nums[i+1]  
        
        for i in range(n):
            prefix[i] = prefix[i] * postfix[i]
        
        return prefix